### PR TITLE
Yaml expansion

### PR DESCRIPTION
**Pull request by perlpunk**
See https://progress.opensuse.org/issues/56876

The original settings hash was modified in the loop.

I sorted all keys in `JobTemplate.pm` to make behaviour reproducible in general.

The fix uses `Storable::dclone()` like discussed in chat to make a deep copy.

(Followup to #2329)

### CODE REVIEW

File: t/api/08-jobtemplates.t [hunk 1]
```diff
@@ -1110,6 +1110,249 @@ subtest 'References' => sub {
...
+        return undef;
+    }
+    $t->get_ok(
+        "/api/v1/job_templates",
+        form => {
+            test_suite_name => 'RAID1',
+            flavor          => ''
+        })->status_is(200);
+    for my $json (@{$t->tx->res->json->{JobTemplates}}) {
+        say Mojo::Util::dumper $json->{settings};
```
Comment (kraih) [inline t/api/08-jobtemplates.t]: This should probably not be in the final test.

Comment (perlpunk) [reply t/api/08-jobtemplates.t]: oops

Comment (perlpunk) [reply t/api/08-jobtemplates.t]: fixed

File: openQA.spec [hunk 2]
```diff
@@ -46,7 +46,7 @@
 %define python_scripts_requires %{nil}
 %endif
 %define assetpack_requires perl(Mojolicious::Plugin::AssetPack) => 1.36, perl(CSS::Minifier::XS) perl(JavaScript::Minifier::XS)
-%define common_requires perl(Config::IniFiles) perl(Cpanel::JSON::XS) perl(Cwd) perl(Data::Dump) perl(Data::Dumper) perl(Digest::MD5) perl(Getopt::Long) perl(Minion) => 9.09, perl(Mojolicious) >= 7.92, perl(Try::Tiny) perl(Regexp::Common)
+%define common_requires perl(Config::IniFiles) perl(Cpanel::JSON::XS) perl(Cwd) perl(Data::Dump) perl(Data::Dumper) perl(Digest::MD5) perl(Getopt::Long) perl(Minion) => 9.09, perl(Mojolicious) >= 7.92, perl(Try::Tiny) perl(Regexp::Common), perl(Storable)
```
Comment (okurz) [inline openQA.spec]: that comma shouldn't be there, only when necessary for more complex statements as in "my-package => 1.0, next"

Comment (perlpunk) [reply openQA.spec]: oh. I saw commas already present in that line, that's why I added it

Comment (Martchus) [reply openQA.spec]: It looks strange that we use the `=>` operator for the Minion version. Shouldn't it be `>=` or is this some strange RPM syntax? (This is of course not related to the change.)

Comment (okurz) [reply openQA.spec]: I don't know, I guess both might work. Wanna try?

**Pull request merged**